src/chainrules.jl

```diff
@@ -111,10 +111,9 @@ end
 
 function ChainRulesCore.rrule(s::Sinus, x::AbstractVector, y::AbstractVector)
     d = x - y
-    sind = sinpi.(d)
-    abs2_sind_r = abs2.(sind) ./ s.r .^ 2
+    abs2_sind_r = (sinpi.(d) ./ s.r) .^ 2
     val = sum(abs2_sind_r)
-    gradx = twoπ .* cospi.(d) .* sind ./ s.r .^ 2
+    gradx = π .* cospi.(2 .* d) ./ s.r .^ 2
```

Comment:
I misremembered the formula 🙈 
```suggestion
    gradx = π .* sinpi.(2 .* d) ./ s.r .^ 2
```

Reply:
No worries!